File: app/Sources/Parley/Contract.swift
import Foundation

// Wire payloads from the Claude Code plugin hooks (see CONTRACT.md).

struct TurnPayload: Codable, Equatable {
    var event: String
    var session_id: String
    var cwd: String
    var project: String
    var tmux_pane: String
    var speak: String
    var label: String?    // spoken project name from .parley.json; nil → use project
    var listen: Bool?     // false = speak-only (no mic/pill); nil/true = speak + listen

    var spokenLabel: String { (label?.isEmpty == false ? label : nil) ?? project }
    var wantsListen: Bool { listen ?? true }
}

// Result of a /turn: the voice reply to inject, and whether the session should PARK
// (stay alive & resumable via /wake) rather than end when the reply is empty.
struct TurnReply { var transcript: String; var park: Bool }

// A paused-but-resumable session, surfaced to the menu and matched by voice command.
struct ParkedInfo: Identifiable, Equatable { var id: String; var label: String; var project: String }

struct ReadyPayload: Codable, Equatable {
    var event: String
    var session_id: String?
    var cwd: String
    var project: String
    var tmux_pane: String
}

enum Contract {
    static func decodeTurn(_ body: Data) -> TurnPayload? {
        try? JSONDecoder().decode(TurnPayload.self, from: body)
    }
    static func decodeReady(_ body: Data) -> ReadyPayload? {
        try? JSONDecoder().decode(ReadyPayload.self, from: body)
    }
}

File: windows/Contract.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Parley;

// Wire payloads from the Claude Code plugin hooks — identical to the macOS contract.
public sealed class TurnPayload
{
    [JsonPropertyName("event")] public string Event { get; set; } = "";
    [JsonPropertyName("session_id")] public string SessionId { get; set; } = "";
    [JsonPropertyName("cwd")] public string Cwd { get; set; } = "";
    [JsonPropertyName("project")] public string Project { get; set; } = "";
    [JsonPropertyName("tmux_pane")] public string TmuxPane { get; set; } = "";
    [JsonPropertyName("speak")] public string Speak { get; set; } = "";
    [JsonPropertyName("label")] public string? Label { get; set; }
    [JsonPropertyName("listen")] public bool? Listen { get; set; }

    [JsonIgnore] public bool WantsListen => Listen ?? true;
    [JsonIgnore] public string SpokenLabel => string.IsNullOrEmpty(Label) ? Project : Label!;

    public static TurnPayload? Decode(string body)
    {
        try { return JsonSerializer.Deserialize<TurnPayload>(body); }
        catch { return null; }
    }
}
